src/utils/RectUtil.ts

```diff
@@ -61,7 +61,18 @@ export class RectUtil {
             }
         }
     }
-
+    static limitRect(rect: IRect) {
```

Comment:
I don't think that function makes a lot of sense on its own

Reply:
sorry my bad takes of reducing complexity. I'll revert that.